core/include/detray/utils/grid/detail/grid_bins.hpp

```diff
@@ -304,6 +326,33 @@ class dynamic_array
         return *this;
     }
 
+    /// Equality operator
+    ///
+    /// @param rhs the bin to be compared with
+    ///
+    /// @returns true if the view is identical
+    DETRAY_HOST_DEVICE
+    constexpr bool operator==(const dynamic_array& rhs) const {
+        // Check if the bin points to the same data
+        if (m_data == rhs.m_data || *m_data == *rhs.m_data) {
```

Comment:
I would switch these two tests around to do the easier check first
```suggestion
        if (*m_data == *rhs.m_data || m_data == rhs.m_data) {
```

Reply:
Isn't the pointer check easier ?